src/main/java/com/github/pereiratostain/generator/MermaidSchemaGenerator.java

```diff
@@ -33,10 +42,90 @@ private void generateHeader(Writer writer) throws IOException {
   }
 
   private void generateEntity(Writer writer, Entity entity) throws IOException {
+    var associations = new ArrayList<String>();
+    var stereotype = "";
+    String fields;
+
+    if(entity.stereotype() == Entity.Stereotype.ENUM) {
+      fields = computeFieldsEnum(entity);
+      stereotype = "\t<<enumeration>>\n";
+    } else {
+      fields = computeFieldsClass(entity, associations);
+    }
+
     writer.append("    class ")
             .append(entity.name())
-            .append(" {")
-            .append("\n    }\n\n");
+            .append(" {\n")
+            .append(stereotype)
+            .append(fields)
+            .append("\n    }\n")
+            .append(generateAssociations(entity, associations))
+            .append("\n");
+  }
+
+  private String computeFieldsEnum(Entity entity) {
+    var fields = new ArrayList<Field>();
+
+    for(var field: entity.fields()) {
+      if(!field.name().equals("$VALUES")) {
+        fields.add(field);
+      }
+    }
+    return generateRecordFields(fields);
+  }
+
+  private String computeFieldsClass(Entity entity, ArrayList<String> associations) {
+    var fields = new ArrayList<Field>();
+    Pattern pattern = Pattern.compile("<.*>");
+
+    for(var field: entity.fields()) {
+      var fieldType = field.type();
+      fieldType = fieldType.replace(";", "");
+
+      Matcher matcher = pattern.matcher(fieldType);
+
+      if(matcher.find()) {
+        var string = matcher.group(0);
+        string = string.replace("<", "");
+        string = string.replace(">", "");
+        fieldType = string;
+      }
+
+      if(entitiesName.contains(fieldType)) {
+        associations.add(fieldType);
+      } else {
+        fields.add(field);
+      }
+    }
+
+    return generateFields(fields);
+  }
+
+  private String generateFields(List<Field> fields) {
+    var string = new StringBuilder();
+
+    for (var field: fields) {
+      string.append("\t").append(modifierToString(field.modifiers().iterator().next())).append(field.name()).append(" : ").append(field.type().replace(";", "")).append("\n");
+    }
+    return string.toString();
+  }
+
+  private String generateRecordFields(List<Field> fields) {
+    var string = new StringBuilder();
+
+    for (var field: fields) {
+      string.append("\t").append(field.name()).append("\n");
+    }
+    return string.toString();
+  }
+
+  private String generateAssociations(Entity entity, List<String> associations) {
+    var string = new StringBuilder();
+
+    for (var field: associations) {
```

Comment:
Use a stream + collect(joining()) to make the code readable.

Reply:
Done